### PR TITLE
Plot settings

### PR DESCRIPTION
**Pull request by joakim-hove**
This is content-wise ready for review.

I intend to polish the commits a bit.

### CODE REVIEW

File: libenkf/src/plot_settings.c [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+
+
+void plot_settings_add_config_items( config_parser_type * config ) {
+  config_schema_item_type * item = config_add_schema_item(config, PLOT_SETTING_KEY, false);
+  config_schema_item_set_argc_minmax(item, 2, 2);
+
+  config_add_key_value(config , PLOT_PATH_KEY         , false , CONFIG_STRING);
+  {
+    char * msg = util_alloc_sprintf( "The keyword %s has been deprecated - use %s %s <PATH>", PLOT_PATH_KEY , PLOT_SETTING_KEY , PATH_KEY );
```
Comment (jepebe) [inline libenkf/src/plot_settings.c]: Should it not be:

`... PLOT_SETTING_KEY, PLOT_PATH_KEY ...`

Comment (jepebe) [inline libenkf/src/plot_settings.c]: Forget it. Saw the additional `%s`

File: python/python/ert/config/schema_item.py [hunk 2]
```diff
@@ -28,11 +30,38 @@ class SchemaItem(BaseCClass):
     _set_argc_minmax = ConfigPrototype("void config_schema_item_set_argc_minmax( schema_item , int , int)")
     _add_alternative = ConfigPrototype("void config_schema_item_add_indexed_alternative(schema_item , int , char*)")
     _set_deprecated = ConfigPrototype("void config_schema_item_set_deprecated(schema_item ,  char*)")
-
+    _valid_string = ConfigPrototype("bool config_schema_item_valid_string(config_content_type_enum ,  char*)", bind = False)
+    _sscanf_bool = UtilPrototype("bool util_sscanf_bool( char* , bool*)", bind = False)
```
Comment (jepebe) [inline python/python/ert/config/schema_item.py]: Is this protyped anywhere else? If yes, then this line will "re-prototype" it. So it is important to ensure the signature is identical or use the originally prototyped function.

File: python/python/ert_gui/plottery/plot_config.py [hunk 3]
```diff
@@ -283,6 +280,6 @@ def copyConfigFrom(self, other):
 
     @classmethod
     def createCopy(cls, other):
-        copy = PlotConfig(None)
+        copy = PlotConfig(PlotSettings( ))
```
Comment (jepebe) [inline python/python/ert_gui/plottery/plot_config.py]: If you do `self._plot_settings = plot_settings` in `__init__` 
then you can do: `copy = PlotConfig(other._plot_settings)` here

File: python/python/ert/config/schema_item.py [hunk 4]
```diff
@@ -28,11 +30,38 @@ class SchemaItem(BaseCClass):
...
         super(SchemaItem, self).__init__(c_ptr)
 
+    @staticmethod
+    def validString(value_type, value):
+        return SchemaItem._valid_string( value_type , value )
+
+        
+    
+    @staticmethod
+    def convert(value_type, value_string):
```
Comment (joakim-hove) [inline python/python/ert/config/schema_item.py]: OK - fixed.

File: python/python/ert/config/schema_item.py (hunk 2 above)
Comment (joakim-hove) [inline python/python/ert/config/schema_item.py]: No - it is not prototyped anywere else. I'll leave it as is now - but I guess the good solution would be to establish a convention for prototyping of loose functions. Technically it is simple, it is mainly the organizing/namespacing I am uncertain of.

**Comment by joakim-hove** [comment]
OK  - some fixup's  pushed. Will rebase and merge if Travis is green.

**Pull request merged**